pkg/discovery/discovery.go

```diff
@@ -114,8 +114,8 @@ func (d *tidbDiscovery) Discover(advertisePeerUrl string) (string, error) {
 		if len(pdAddresses) != 0 {
 			return fmt.Sprintf("--join=%s", strings.Join(pdAddresses, ",")), nil
 		}
-		// Initialize the PD cluster with the FQDN format service record if tc.Spec.ClusterDomain is set.
-		if len(tc.Spec.ClusterDomain) > 0 {
+		// Initialize the PD cluster with the FQDN format service record if deploy across k8s.
+		if tc.AcrossK8s() {
```

Comment:
This change is incorrect for the case where `tc.Spec.ClusterDomain` is not set in the across Kubernetes case, the original check is correct.